src/views/components/input/file/InputFile.native.js

```diff
@@ -1,49 +1,208 @@
+    this.setState( prevState => ({
+      files: [
+        ...prevState.files,
+        result,
+      ],
+    }), () => {
+      this.handleSaveToServer;
+    });
+  }
+
+  handleSaveToServer = () => {
+    const { files } = this.state;
+    
+    if ( this.props.onChange ) {
+      this.props.onChange({ target: { value: files } });
+    }
+  }
+
+  uploadFile = result => {
+    const localUri = result.uri;
+    const filename = localUri.split( '/' ).pop();
+    const match = /\.(\w+)$/.exec( filename );
+
+    const type = match ? mime.contentType( match[1] ).split( ';' )[0] : 'file';  
+    const formData = new FormData();
+    const url = 'https://uppych40.channel40.com.au/s3/';
+    
+    axios({
+      method: 'get',
+      url: `${url}params?filename=${filename}&type=${type.split( '/' )[0]}%2F${type.split( '/' )[1]}`,
+    })
+    .then( response => {
+      if ( response.status === 200 ) {
+        if ( dlv( response, 'data.fields' )) {
+          const data = response.data;
+          const fields = response.data.fields;
+          
+          Object.keys( fields ).forEach( field_key => {
+            formData.append( field_key, fields[field_key] );
+          });
+
+          formData.append( 'file', { uri: localUri, name: filename, type });
+          
+          fetch( response.data.url, {
+            method: 'POST',
+            body: formData,
+            header: {
+              'content-type': 'multipart/form-data',
+            },
+          })
+          .then( async response => {
+            const text = await response.text();
+
+            const jsonObj = fastXmlParser.parse( text );
+            
+            const formattedFile = {
+              data: {
+                name: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].name,
+                // size: 65240,
+                type: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].type,
+              },
+              extension: type.split( '/' )[1],
+              id: dlv( jsonObj, 'PostResponse.Key' ),
+              meta: {
+                ...fields,
+                name: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].name,
+                type: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].type,
+              },
+              name: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].name,
+              response: {
+                body: jsonObj.PostResponse,
+                status: fields.success_action_status,
+                uploadURL: dlv( jsonObj, 'PostResponse.Location' ),
+              },
+              // size: ,
+              type: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].type,
+              uploadURL: dlv( jsonObj, 'PostResponse.Location' ),
+              uploaded: true,
+              xhrUpload: {
+                endpoint: data.url,
+                formData: true,
+                metaFields: Object.keys( fields ).map( field_key => {
```

Comment:
The use of `.map()` here seems redundant, are we able to just use `Object.keys( fields )`?